### ISSUE TITLE
Can't vertical scroll page on mobile devices with full height owl carousel

### ISSUE THREAD

**Issue by lholdingfoundry** [OwlCarousel2/OwlCarousel2#2143]
I have a 100vh height owl carousel slider in my header on mobile (Android/Chrome) but can't swipe to scroll down the page (touch event on top of slider). Side swipe to trigger slide transitions works fine but no page scrolling is happening on upward or downward swipe. Using version 2.3.0.

Any help appreciated! Thanks!

**Comment by martinkariuki7** [comment]
I have the same issue

**Comment by martinkariuki7** [comment]
I worked around the mobile issue by setting mouseDrag to false and touchDrag to true.

**Comment by anilljoshi** [comment]
could you please share the code

**Comment by denisbosire** [comment]
@martinkariuki7 solution helped.

**Comment by ExillustX** [comment]
any news about this issue?

**Comment by exil0867** [comment]
After 4 hours of trying to fix this issue... i finally found a fix, using this [version 2.2.1 ](https://owlcarousel2.github.io/OwlCarousel2/assets/owlcarousel/assets/owl.carousel.min.css)of owlCarousel CSS, seems to a fix, so this issue is not serious at all, just a buggy css file. i'm also not sure about the disadvantages of using v2.2.1 CSS file along with v2.3.0 JS file. whatever, someone need to find out which CSS property in version 2.3.0 causing the issue, then fix it and send a push request, sorry but i don't have time for this. 
Update: Instead of replacing the whole CSS file to an older version, i made it easy for you, you can just add this code to your CSS and the issue is fixed.
Code: 
```
.owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
    -ms-touch-action: auto;
        touch-action: auto;
}
```

**Comment by exil0867** [comment]
Also @martinkariuki7 's answer is not a good approach since disabling touchDrag/mouseDrag will affect all the devices, and this issue is about mobile devices with touch screens, i know you can achieve it by sniffing the user agent, but sniffing browser's UA in 2018 is the worst idea i guess.

**Comment by spandan888** [comment]
@ExillustX that solves the issue. thanks!

**Comment by gecsegdaniel** [comment]
It worked for me in all device (mobile, tablet, PC). Based on @martinkariuki7 's answer. Standard setting for mouseDrag is false and for touchDrag is true. These need to changed on touch devices.
`$(SELECTOR).owlCarousel({
        loop: false,
        margin: 15,
        nav: true,
        mouseDrag: true,
        touchDrag: false,
        navContainer: 'controls-class',
        dots: false,
        navText: ['<span class="icon-arrow-left"></span>', '<span class="icon-arrow-right"></span>'],
        responsive: {
          0: {
            items: 1,
            autoHeight: true,
            mouseDrag: false,
            touchDrag: true
          },
          600: {
            items: 1,
            autoHeight: true,
            mouseDrag: false,
            touchDrag: true
          },
          1050: {
            items: 3,
            autoWidth: true
          }
        }
      });`

**Comment by pascalporedda** [comment]
(Sorry about the canned reply.)

You have not added a link to a working live example. See [CONTRIBUTING.md](https://github.com/OwlCarousel2/OwlCarousel2/blob/develop/CONTRIBUTING.md).

Please add a link to a page where I can verify this error. You can use a service like [jsfiddle](http://jsfiddle.net/), [plnkr](https://plnkr.co/), [jsbin](http://jsbin.com/) or [codepen](http://codepen.io/).

And make sure you're using the latest Owl Version!

**Comment by pascalporedda** [comment]
As far as I can see this is related to using a full screen slider, if that is the case the answer is already proposed by ExillustX simply override the css touch-action to auto or pan-y

**Comment by rohankhudedev** [comment]
For version 2.3.3: Above CSS fixed by issue. But by setting mouseDrag to false and touchDrag to true it didn't worked.

**Comment by gleenk** [comment]
In 2.3.3 is still bugged. In 2.3.4 it works.

**Comment by ritsrivastava01** [comment]
Thx @ExillustX its fixed my issue as well. 👍

**Comment by realaaz** [comment]
@ExillustX's fix works for me in iOS & Android, on Safari, Chrome, and native browsers. Thank you a lot! 👍

**Comment by boesbo** [comment]
> @ExillustX's fix works for me in iOS & Android, on Safari, Chrome, and native browsers. Thank you a lot! 👍

On iOS it's impossible.

**Comment by jkhhihi** [comment]
It is not working.
Still it is up and down on mobile device.
Even owlCarousel website slide
check it out  [https://owlcarousel2.github.io/OwlCarousel2/](https://owlcarousel2.github.io/OwlCarousel2/)
on your mobile phone.
When you swipe, the screen is up and down as you see OwlCarousel website slides.
(Even 2.3.4 that i tried)

**Comment by nuriye** [comment]
I am using 2.3.3 - the css variation did not work, but the js did. But on another Website the css variation worked. Strange. :) But thank you!

**Comment by koredeadebayo** [comment]
Thanks 

> After 4 hours of trying to fix this issue... i finally found a fix, using this [version 2.2.1 ](https://owlcarousel2.github.io/OwlCarousel2/assets/owlcarousel/assets/owl.carousel.min.css)of owlCarousel CSS, seems to a fix, so this issue is not serious at all, just a buggy css file. i'm also not sure about the disadvantages of using v2.2.1 CSS file along with v2.3.0 JS file. whatever, someone need to find out which CSS property in version 2.3.0 causing the issue, then fix it and send push request, sorry but i don't have time for this. ;_;
> Update: Instead of replacing the whole CSS file to an older version, i made it easy for you, you can just add this code to your CSS and the issue is fixed.
> Code:
> 
> ```
> .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
>     -ms-touch-action: auto;
>         touch-action: auto;
> }
> ```

It worked for me. Thanks

**Comment by fikar46** [comment]
> I worked around the mobile issue by setting mouseDrag to false and touchDrag to true.

this is work for me

`var carousel = function() {
		$('.mypage-slider').owlCarousel({
	    loop:true,
	    autoplay: true,
	    margin:0,
	    animateOut: 'fadeOut',
	    animateIn: 'fadeIn',
	    nav:false,
	    autoplayHoverPause: false,
			items: 1,
			touchDrag:true,
			mouseDrag:false,
	    navText : ["<span class='ion-md-arrow-back'></span>","<span class='ion-chevron-right'></span>"],
	    responsive:{
	      0:{
	        items:1
	      },
	      600:{
	        items:1
	      },
	      1000:{
	        items:1
	      }
	    }
		});
	};
	carousel();`

**Comment by yasineryigit** [comment]
> After 4 hours of trying to fix this issue... i finally found a fix, using this [version 2.2.1 ](https://owlcarousel2.github.io/OwlCarousel2/assets/owlcarousel/assets/owl.carousel.min.css)of owlCarousel CSS, seems to a fix, so this issue is not serious at all, just a buggy css file. i'm also not sure about the disadvantages of using v2.2.1 CSS file along with v2.3.0 JS file. whatever, someone need to find out which CSS property in version 2.3.0 causing the issue, then fix it and send push request, sorry but i don't have time for this. ;_;
> Update: Instead of replacing the whole CSS file to an older version, i made it easy for you, you can just add this code to your CSS and the issue is fixed.
> Code:
> 
> ```
> .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
>     -ms-touch-action: auto;
>         touch-action: auto;
> }
> ```

It worked. Thanks a lot :)

**Comment by AjithPG** [comment]
@ExillustX  Thanks for your solution it worked.You are awsome

**Comment by doddannad** [comment]
> After 4 hours of trying to fix this issue... i finally found a fix, using this [version 2.2.1 ](https://owlcarousel2.github.io/OwlCarousel2/assets/owlcarousel/assets/owl.carousel.min.css)of owlCarousel CSS, seems to a fix, so this issue is not serious at all, just a buggy css file. i'm also not sure about the disadvantages of using v2.2.1 CSS file along with v2.3.0 JS file. whatever, someone need to find out which CSS property in version 2.3.0 causing the issue, then fix it and send push request, sorry but i don't have time for this. ;_;
> Update: Instead of replacing the whole CSS file to an older version, i made it easy for you, you can just add this code to your CSS and the issue is fixed.
> Code:
> 
> ```
> .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
>     -ms-touch-action: auto;
>         touch-action: auto;
> }
> ```

thank you soo much you did greate job

**Comment by xarala221** [comment]
> It worked for me in all device (mobile, tablet, PC). Based on @martinkariuki7 's answer. Standard setting for mouseDrag is false and for touchDrag is true. These need to changed on touch devices.
> 
> $(SELECTOR).owlCarousel({
>         loop: false,
>         margin: 15,
>         nav: true,
>         mouseDrag: true,
>         touchDrag: false,
>         navContainer: 'controls-class',
>         dots: false,
>         navText: ['', ''],
>         responsive: {
>           0: {
>             items: 1,
>             autoHeight: true,
>             mouseDrag: false,
>             touchDrag: true
>           },
>           600: {
>             items: 1,
>             autoHeight: true,
>             mouseDrag: false,
>             touchDrag: true
>           },
>           1050: {
>             items: 3,
>             autoWidth: true
>           }
>         }
>       });

This works for me!

**Comment by rolfo85** [comment]
To me this didn't work

`
    .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
        -ms-touch-action: auto;
        touch-action: auto;
    }
`

but this did work

`
    .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item, .owl-carousel{
        -ms-touch-action: auto;
        touch-action: auto;
    }
`

**Comment by codegreenie** [comment]
> .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
>     -ms-touch-action: auto;
>         touch-action: auto;
> }

This is a Life Saver. Thank you very much

**Comment by harryliangye** [comment]
> After 4 hours of trying to fix this issue... i finally found a fix, using this [version 2.2.1 ](https://owlcarousel2.github.io/OwlCarousel2/assets/owlcarousel/assets/owl.carousel.min.css)of owlCarousel CSS, seems to a fix, so this issue is not serious at all, just a buggy css file. i'm also not sure about the disadvantages of using v2.2.1 CSS file along with v2.3.0 JS file. whatever, someone need to find out which CSS property in version 2.3.0 causing the issue, then fix it and send a push request, sorry but i don't have time for this.
> Update: Instead of replacing the whole CSS file to an older version, i made it easy for you, you can just add this code to your CSS and the issue is fixed.
> Code:
> 
> ```
> .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
>     -ms-touch-action: auto;
>         touch-action: auto;
> }
> ```

That worked really good for me! 

Many many Thanks!

**Comment by Sihap06** [comment]
thx. its work for me

**Comment by Eblazecode** [comment]
thanks @martinkariuki7. This really worked for me . You saved from a whole lot of stress
.owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
    -ms-touch-action: auto;
        touch-action: auto;
}

**Comment by aafmj** [comment]
> After 4 hours of trying to fix this issue... i finally found a fix, using this [version 2.2.1 ](https://owlcarousel2.github.io/OwlCarousel2/assets/owlcarousel/assets/owl.carousel.min.css)of owlCarousel CSS, seems to a fix, so this issue is not serious at all, just a buggy css file. i'm also not sure about the disadvantages of using v2.2.1 CSS file along with v2.3.0 JS file. whatever, someone need to find out which CSS property in version 2.3.0 causing the issue, then fix it and send a push request, sorry but i don't have time for this.
> Update: Instead of replacing the whole CSS file to an older version, i made it easy for you, you can just add this code to your CSS and the issue is fixed.
> Code:
> 
> ```
> .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
>     -ms-touch-action: auto;
>         touch-action: auto;
> }
> ```

Thank you !! it works..

**Comment by codebysanjith** [comment]
> 
> 
> After 4 hours of trying to fix this issue... i finally found a fix, using this [version 2.2.1 ](https://owlcarousel2.github.io/OwlCarousel2/assets/owlcarousel/assets/owl.carousel.min.css)of owlCarousel CSS, seems to a fix, so this issue is not serious at all, just a buggy css file. i'm also not sure about the disadvantages of using v2.2.1 CSS file along with v2.3.0 JS file. whatever, someone need to find out which CSS property in version 2.3.0 causing the issue, then fix it and send a push request, sorry but i don't have time for this.
> Update: Instead of replacing the whole CSS file to an older version, i made it easy for you, you can just add this code to your CSS and the issue is fixed.
> Code:
> 
> ```
> .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
>     -ms-touch-action: auto;
>         touch-action: auto;
> }
> ```

Thanks Buddy <3

**Comment by exil0867** [comment]
No problem guys. Glad I was helpful!

**Comment by SurajanShrestha** [comment]
> I worked around the mobile issue by setting mouseDrag to false and touchDrag to true.

@martinkariuki7 Thanks Man...This Solution Worked :)

**Comment by daviddagb2** [comment]
> .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
>     -ms-touch-action: auto;
>         touch-action: auto;
> }

This answer help me a lot, thanks

**Comment by anubhav15896** [comment]
> After 4 hours of trying to fix this issue... i finally found a fix, using this [version 2.2.1 ](https://owlcarousel2.github.io/OwlCarousel2/assets/owlcarousel/assets/owl.carousel.min.css)of owlCarousel CSS, seems to a fix, so this issue is not serious at all, just a buggy css file. i'm also not sure about the disadvantages of using v2.2.1 CSS file along with v2.3.0 JS file. whatever, someone need to find out which CSS property in version 2.3.0 causing the issue, then fix it and send a push request, sorry but i don't have time for this.
> Update: Instead of replacing the whole CSS file to an older version, i made it easy for you, you can just add this code to your CSS and the issue is fixed.
> Code:
> 
> ```
> .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
>     -ms-touch-action: auto;
>         touch-action: auto;
> }
> ```

This Works for me. Thanks ExillustX.

**Comment by yrodelo** [comment]
@ExillustX 2020 and your solution it is still the best and simplest solution thanks!

**Comment by ioktar** [comment]
> 4 saat bu sorunu çözmeye çalıştıktan sonra ... nihayet bir düzeltme buldum , owlCarousel CSS'nin bu [2.2.1 sürümünü](https://owlcarousel2.github.io/OwlCarousel2/assets/owlcarousel/assets/owl.carousel.min.css) kullanarak bir düzeltme gibi görünüyor, bu nedenle bu sorun hiç de ciddi değil, sadece hatalı bir css dosyası. v2.2.1 CSS dosyasını v2.3.0 JS dosyası ile birlikte kullanmanın dezavantajlarından da emin değilim. her neyse, birisinin soruna 2.3.0 sürümünde hangi CSS özelliğinin neden olduğunu bulması, ardından düzeltmesi ve bir push isteği göndermesi gerekiyor, üzgünüm ama bunun için zamanım yok.
> Güncelleme: Tüm CSS dosyasını eski bir versiyonla değiştirmek yerine, sizin için kolaylaştırdım, sadece bu kodu CSS'nize ekleyebilirsiniz ve sorun giderildi.
> Kod:
> 
> ```
> .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
>     -ms-touch-action: auto;
>         touch-action: auto;
> }
> ```

solution is successful. Thank you
```
.owl-carousel.owl-drag .owl-item{
-ms-touch-action:auto;touch-action:auto;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}
```

**Comment by possibility-of-offense** [comment]
I will post my solution here as well; I'd had the same problem and I fixed it by targeting .owl-stage and setting the pointer-events to none.
.owl-stage {
    pointer-events: none !important;
}
The problem is that in this way you are gonna lose the drag functionality. You can still switch the slides through the navigational arrows though. I think it is not the best solution but when you are in need, it works fine.

**Comment by apunteserrantes** [comment]
In my case, the solution that worked was:

.owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item {
    -ms-touch-action: pan-y !important;
    touch-action: pan-y !important;
}

**Comment by siegelion** [comment]
> I worked around the mobile issue by setting mouseDrag to false and touchDrag to true.

thanks ,very very thanks ,its been bothering me for long time;

**Comment by SantiagoVagni** [comment]
> After 4 hours of trying to fix this issue... i finally found a fix, using this [version 2.2.1 ](https://owlcarousel2.github.io/OwlCarousel2/assets/owlcarousel/assets/owl.carousel.min.css)of owlCarousel CSS, seems to a fix, so this issue is not serious at all, just a buggy css file. i'm also not sure about the disadvantages of using v2.2.1 CSS file along with v2.3.0 JS file. whatever, someone need to find out which CSS property in version 2.3.0 causing the issue, then fix it and send a push request, sorry but i don't have time for this. Update: Instead of replacing the whole CSS file to an older version, i made it easy for you, you can just add this code to your CSS and the issue is fixed. Code:
> 
> ```
> .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
>     -ms-touch-action: auto;
>         touch-action: auto;
> }
> ```

How are you? I have a problem with that solution, when i use it in LocalHost it perfectly works, but when i use it in a web page host it dosnt work. A solution??

**Comment by exil0867** [comment]
> > After 4 hours of trying to fix this issue... i finally found a fix, using this [version 2.2.1 ](https://owlcarousel2.github.io/OwlCarousel2/assets/owlcarousel/assets/owl.carousel.min.css)of owlCarousel CSS, seems to a fix, so this issue is not serious at all, just a buggy css file. i'm also not sure about the disadvantages of using v2.2.1 CSS file along with v2.3.0 JS file. whatever, someone need to find out which CSS property in version 2.3.0 causing the issue, then fix it and send a push request, sorry but i don't have time for this. Update: Instead of replacing the whole CSS file to an older version, i made it easy for you, you can just add this code to your CSS and the issue is fixed. Code:
> > 
> > ```
> > .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
> >     -ms-touch-action: auto;
> >         touch-action: auto;
> > }
> > ```
> 
> How are you? I have a problem with that solution, when i use it in LocalHost it perfectly works, but when i use it in a web page host it dosnt work. A solution?? 

Try hard refreshing the page to clear its cache by pressing ctrl+shift+r.

**Comment by SantiagoVagni** [comment]
> > > After 4 hours of trying to fix this issue... i finally found a fix, using this [version 2.2.1 ](https://owlcarousel2.github.io/OwlCarousel2/assets/owlcarousel/assets/owl.carousel.min.css)of owlCarousel CSS, seems to a fix, so this issue is not serious at all, just a buggy css file. i'm also not sure about the disadvantages of using v2.2.1 CSS file along with v2.3.0 JS file. whatever, someone need to find out which CSS property in version 2.3.0 causing the issue, then fix it and send a push request, sorry but i don't have time for this. Update: Instead of replacing the whole CSS file to an older version, i made it easy for you, you can just add this code to your CSS and the issue is fixed. Code:
> > > ```
> > > .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
> > >     -ms-touch-action: auto;
> > >         touch-action: auto;
> > > }
> > > ```
> > 
> > 
> > How are you? I have a problem with that solution, when i use it in LocalHost it perfectly works, but when i use it in a web page host it dosnt work. A solution??
> 
> Try hard refreshing the page to clear its cache by pressing ctrl+shift+r.

Omg, Thankss!!!!

**Comment by nandha46** [comment]
> After 4 hours of trying to fix this issue... i finally found a fix, using this [version 2.2.1 ](https://owlcarousel2.github.io/OwlCarousel2/assets/owlcarousel/assets/owl.carousel.min.css)of owlCarousel CSS, seems to a fix, so this issue is not serious at all, just a buggy css file. i'm also not sure about the disadvantages of using v2.2.1 CSS file along with v2.3.0 JS file. whatever, someone need to find out which CSS property in version 2.3.0 causing the issue, then fix it and send a push request, sorry but i don't have time for this. Update: Instead of replacing the whole CSS file to an older version, i made it easy for you, you can just add this code to your CSS and the issue is fixed. Code:
> 
> ```
> .owl-carousel .owl-stage, .owl-carousel.owl-drag .owl-item{
>     -ms-touch-action: auto;
>         touch-action: auto;
> }
> ```

Thanks. This is helpful.

**Comment by exil0867** [comment]
You are welcome.
